src/utils.c

```diff
@@ -353,7 +353,7 @@ int map_intd(char *s, char **v, int dv)
 		return atoi(s);
 	}
 	for (i = 0; v[i]; i++)
-		if (!strncasecmp(s, v[i], strlen(v[i])))
+		if (strlen(s) == strlen(v[i]) && !strncasecmp(s, v[i], strlen(v[i])))
```

Comment:
Not sure this is a good change. The whole point of mapintd is to check the first part of both strings. There are plenty of places that can happen. Of course the new line is equivalent of strcasecmp.

Where is this check going wrong currently?

Reply:
Hi @catalinii ,

> Where is this check going wrong currently?

I commented previously. Even though, I repeat it: 😉 
- When the name of two _different_ modules have a collission, the incorrect module is selected. You can test it enabling only the debug of the module `socketworks`. So, as the module `socket` is a substring of the `socketworks`, and it's listed before, the code selects the module "socket" when you request to use "socketworks". So instead of changing the name of a module (as I feel you don't want to do it), I provide this definitive patch.

I'm surprissed you never see this before!
And, why you want to use "substrings" and not the full name?

Reply:
Hi @catalinii ,

No feedback about my last comment? Why you want to support substrings? And what you prefer to fix the current bug: change the name of the "socketworks" module or disable the substrings support?

Please, note that bug exists and we need to resolve it!

Reply:
I think the best way is to create a new function that uses strcasecmp and use that function instead of map_intd.

Reply:
Hi @catalinii ,

> I think the best way is to create a new function that uses strcasecmp and use that function instead of map_intd.

Ok. Now I see the point: the `map_intd()` function is used for other uses too. So, _only_ for the detection of the modules name here: https://github.com/catalinii/minisatip/blob/da53e5a9b26da9bd28fc71aa63958e5635fa1cc1/src/minisatip.c#L681 it will be necessary to use another different (aka. new) function.

That's your opinion?

Reply:
Yeah, create a new function: string_to_index

Reply:
@lars18th the previous yeah was not applying to your last comment.

Basically a new function is a better option as loose matching is not a good idea in general as it can create such issues you are pointing out. I would rather migrate some map_intd use cases to the new function. The other problem with this function is that it acts also as atoi.

I think creating a function that just gets the index of a string in an array makes more sense.

Once is done you can try to add some unit tests in test_utils.c to make sure it does what you want.

Reply:
Hi @catalinii ,

Thank you for the response. So the `map_intd()` requires to be reimplemented with a new name. OK. However, I don't have sufficient time now to do all tasks that you request. So, please, as the `<=` check doesn't create new side effects, and solves the bug of "socketworks" logging, then you will be so kind as to accept this PR?

Regards.

Reply:
Hi @catalinii ,

(ping) Are you confortable with the current (the `<=` check) workaround that resolves the _current_ bug? In the meantime I feel is preferable to use it.

Please, comment it.
Thank you!

Reply:
The use cases I mention that the function will break is exactly when strlen(s) > strlen(s[i]).

With the current method you cannot fix both cases: the bug in LOG and the issues I mention about. You need 2 methods to fix both of them.
map_intd is used intensively to parse the stream parameters (like msys, mod, ... ). Breaking that part will make minisatip  unusable that's why I am reluctant to accept the patch as it is.

Reply:
> The use cases I mention that the function will break is exactly when strlen(s) > strlen(s[i]).
> 
> With the current method you cannot fix both cases: the bug in LOG and the issues I mention about. You need 2 methods to fix both of them.
> map_intd is used intensively to parse the stream parameters (like msys, mod, ... ). Breaking that part will make minisatip unusable that's why I am reluctant to accept the patch as it is.

OK, now I understand the reason. So, the only option is to create or use another different function when checking the LOG. 👍